src/rgw/store/dbstore/common/dbstore.cc

```diff
@@ -1968,5 +1988,72 @@ int DB::put_head(const std::string& oid, const rgw::sal::Lifecycle::LCHead& head
   return ret;
 }
 
+int DB::delete_stale_objs(const DoutPrefixProvider *dpp, const std::string& bucket) {
+  DBOpParams params = {};
+  int ret = -1;
+
+  params.op.bucket.info.bucket.name = bucket;
+  /* Verify if bucket exists.
+   * XXX: This is needed for now to create objectmap of bucket
+   * in SQLGetBucket
+   */
+  InitializeParams(dpp, "GetBucket", &params);
+  ret = ProcessOp(dpp, "GetBucket", &params);
+  if (ret) {
+    ldpp_dout(dpp, 0) << "In GetBucket failed err:(" <<ret<<")" << dendl;
+  }
+
+  /* XXX: handle reads racing with delete here. Simple approach is maybe
+   * to use locks or sqlite transactions.
+   */
```

Comment:
certainly we should not be able to crash;  I think logically, we need whatever we do to be visible in the database, if we want to be general?

Reply:
yes.

Reply:
This is now handled as part of the latest commit in this PR.